controllers/drcluster_controller.go

```diff
@@ -91,7 +92,8 @@ const (
 func (r *DRClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	// TODO: Validate managedCluster name? and also ensure it is not deleted!
 	// TODO: Setup views for storage class and VRClass to read and report IDs
-	log := ctrl.Log.WithName("controllers").WithName("drcluster").WithValues("name", req.NamespacedName.Name)
+	log := ctrl.Log.WithName("controllers").WithName("drcluster").WithValues("name",
+		req.NamespacedName.Name, "Rid", uuid.New())
```

Comment:
It will be easier to follow if we use:

    log := ctrl.Log.WithName("controllers")
                   .WithName("drcluster")
                   .WithValues("name", req.NamespacedName.Name, "Rid", uuid.New())

What is the meaning of "Rid"? is this common value documented somewhere? If we don't have to use this value, it would be more consistent to use "rid" since other names/tags are lower case.

Can you share the output of the log with this change? Adding it to the commit message will make the change more clear and easier to review.

Reply:
Rid is Reconciler id which will be generated for each reconcile. It helps to track each reconcile. 
For example I have pasted log for drpolicy for an entire reconcile. 
Drpolicy log2022-08-06T20:44:17.722+0530    INFO    controllers.drpolicy    reconcile enter {"name": "my-async-dr-peers", "rid": "ad1b0695-9929-45a3-bc39-1d421fdfbc9c"}
2022-08-06T20:44:17.723+0530    INFO    controllers.drcluster   Creating ManifestWork   {"name": "west1-cluster", "rid": "4e0f5f09-d962-4bda-a5e8-4c7b3702e5e8", "cluster": "west1-cluster", "MW": {"namespace": "west1-cluster", "name": "ramen-dr-cluster"}}
2022-08-06T20:44:17.723+0530    INFO    controllers.drpolicy    create/update   {"name": "my-async-dr-peers", "rid": "ad1b0695-9929-45a3-bc39-1d421fdfbc9c"}
2022-08-06T20:44:17.724+0530    INFO    controllers.drpolicy    finalizer or label add  {"name": "my-async-dr-peers", "rid": "ad1b0695-9929-45a3-bc39-1d421fdfbc9c"}
2022-08-06T20:44:17.726+0530    INFO    controllers.drpolicy    Add Secret      {"name": "my-async-dr-peers", "rid": "ad1b0695-9929-45a3-bc39-1d421fdfbc9c", "cluster": "east1-cluster", "secret": "s3secret0"}
2022-08-06T20:44:17.726+0530    INFO    controllers.drcluster   Phase: Current ''. Next 'Available'     {"name": "west1-cluster", "rid": "4e0f5f09-d962-4bda-a5e8-4c7b3702e5e8"}
2022-08-06T20:44:17.727+0530    INFO    controllers.drpolicy    Creating secret policy  {"name": "my-async-dr-peers", "rid": "ad1b0695-9929-45a3-bc39-1d421fdfbc9c", "secret": "s3secret0", "cluster": "east1-cluster", "namespace": "ns-envtest"}
2022-08-06T20:44:17.730+0530    INFO    controllers.drcluster   Updated drCluster Status {Phase:Available Conditions:[{Type:Fenced Status:False ObservedGeneration:1 LastTransitionTime:2022-08-06 20:44:17 +0530 IST Reason:Clean Message:Cluster Clean} {Type:Clean Status:True ObservedGeneration:1 LastTransitionTime:2022-08-06 20:44:17 +0530 IST Reason:Clean Message:Cluster Clean} {Type:Validated Status:True ObservedGeneration:1 LastTransitionTime:2022-08-06 20:44:17 +0530 IST Reason:Succeeded Message:Validated the cluster}]} {"name": "west1-cluster", "rid": "4e0f5f09-d962-4bda-a5e8-4c7b3702e5e8"}
2022-08-06T20:44:17.730+0530    INFO    controllers.drcluster   reconcile exit  {"name": "west1-cluster", "rid": "4e0f5f09-d962-4bda-a5e8-4c7b3702e5e8"}
2022-08-06T20:44:17.730+0530    INFO    controllers.drcluster   reconcile enter {"name": "east1-cluster", "rid": "2a9b38b7-ff8f-4836-8377-70a10e800e12"}
2022-08-06T20:44:17.730+0530    INFO    controllers.drcluster   Phase: Current ''. Next 'Starting'      {"name": "east1-cluster", "rid": "2a9b38b7-ff8f-4836-8377-70a10e800e12"}
2022-08-06T20:44:17.731+0530    INFO    controllers.drpolicy    Initializing secret policy trigger      {"name": "my-async-dr-peers", "rid": "ad1b0695-9929-45a3-bc39-1d421fdfbc9c", "secret": "s3secret0", "trigger": "1312"}
2022-08-06T20:44:17.732+0530    INFO    controllers.drcluster   create/update   {"name": "east1-cluster", "rid": "2a9b38b7-ff8f-4836-8377-70a10e800e12"}
2022-08-06T20:44:17.732+0530    INFO    controllers.drcluster   finalizer or label add  {"name": "east1-cluster", "rid": "2a9b38b7-ff8f-4836-8377-70a10e800e12"}
2022-08-06T20:44:17.735+0530    INFO    controllers.drpolicy    Add Secret      {"name": "my-async-dr-peers", "rid": "ad1b0695-9929-45a3-bc39-1d421fdfbc9c", "cluster": "west1-cluster", "secret": "s3secret0"}
2022-08-06T20:44:17.735+0530    INFO    controllers.drcluster   Creating ManifestWork   {"name": "east1-cluster", "rid": "2a9b38b7-ff8f-4836-8377-70a10e800e12", "cluster": "east1-cluster", "MW": {"namespace": "east1-cluster", "name": "ramen-dr-cluster"}}
2022-08-06T20:44:17.736+0530    INFO    controllers.drpolicy    Updating placement rule for secret policy       {"name": "my-async-dr-peers", "rid": "ad1b0695-9929-45a3-bc39-1d421fdfbc9c", "secret": "s3secret0", "clusters": [{"name":"east1-cluster"},{"name":"west1-cluster"}]}
2022-08-06T20:44:17.738+0530    INFO    controllers.drpolicy    condition append        {"name": "my-async-dr-peers", "rid": "ad1b0695-9929-45a3-bc39-1d421fdfbc9c", "type": "Validated", "status": "True", "reason": "Succeeded", "message": "drpolicy validated", "generation": 1}
2022-08-06T20:44:17.738+0530    INFO    controllers.drcluster   Phase: Current ''. Next 'Available'     {"name": "east1-cluster", "rid": "2a9b38b7-ff8f-4836-8377-70a10e800e12"}
2022-08-06T20:44:17.740+0530    INFO    controllers.drpolicy    reconcile exit  {"name": "my-async-dr-peers", "rid": "ad1b0695-9929-45a3-bc39-1d421fdfbc9c"}


DRcluster log
2022-08-06T20:43:14.318+0530    INFO    controller.drcluster    Starting workers        {"reconciler group": "ramendr.openshift.io", "reconciler kind": "DRCluster", "worker count": 1}
2022-08-06T20:43:14.318+0530    INFO    controllers.drcluster   reconcile enter {"name": "drc-cluster0", "rid": "9dad5b86-8b29-49ea-958a-f9b4bfaef6f6"}
2022-08-06T20:43:14.318+0530    INFO    controllers.drcluster   Phase: Current ''. Next 'Starting'      {"name": "drc-cluster0", "rid": "9dad5b86-8b29-49ea-958a-f9b4bfaef6f6"}
2022-08-06T20:43:14.319+0530    INFO    controller.volumereplicationgroup       Starting workers        {"reconciler group": "ramendr.openshift.io", "reconciler kind": "VolumeReplicationGroup", "worker count": 1}
2022-08-06T20:43:14.319+0530    INFO    controllers.drcluster   create/update   {"name": "drc-cluster0", "rid": "9dad5b86-8b29-49ea-958a-f9b4bfaef6f6"}
2022-08-06T20:43:14.319+0530    INFO    controllers.drcluster   finalizer or label add  {"name": "drc-cluster0", "rid": "9dad5b86-8b29-49ea-958a-f9b4bfaef6f6"}
2022-08-06T20:43:14.319+0530    INFO    controller.drpolicy     Starting workers        {"reconciler group": "ramendr.openshift.io", "reconciler kind": "DRPolicy", "worker count": 1}
2022-08-06T20:43:14.319+0530    INFO    controller.drplacementcontrol   Starting workers        {"reconciler group": "ramendr.openshift.io", "reconciler kind": "DRPlacementControl", "worker count": 1}
2022-08-06T20:43:14.319+0530    INFO    controllers.drpolicy    reconcile enter {"name": "sync-drpolicy-drcluster-tests", "rid": "5b3daefa-cefc-4833-829b-80f5b28ac52a"}
2022-08-06T20:43:14.320+0530    INFO    controllers.drpolicy    create/update   {"name": "sync-drpolicy-drcluster-tests", "rid": "5b3daefa-cefc-4833-829b-80f5b28ac52a"}
2022-08-06T20:43:14.322+0530    INFO    controllers.drpolicy    finalizer or label add  {"name": "sync-drpolicy-drcluster-tests", "rid": "5b3daefa-cefc-4833-829b-80f5b28ac52a"}
2022-08-06T20:43:14.323+0530    INFO    controllers.drcluster   Creating ManifestWork   {"name": "drc-cluster0", "rid": "9dad5b86-8b29-49ea-958a-f9b4bfaef6f6", "cluster": "drc-cluster0", "MW": {"namespace": "drc-cluster0", "name": "ramen-dr-cluster"}}

Volumereplicationgroup log
2022-08-06T20:43:55.437+0530    INFO    controllers.VolumeReplicationGroup      Updating VRG status     {"VolumeReplicationGroup": "default/asdf", "rid": "62208185-4b37-4fb6-b12d-8df78fbf58a8"}
2022-08-06T20:43:55.437+0530    DEBUG   events  Warning {"object": {"kind":"VolumeReplicationGroup","namespace":"default","name":"asdf","uid":"7d099716-ec3a-4aa6-bb76-426ff15bd221","apiVersion":"ramendr.openshift.io/v1alpha1","resourceVersion":"421"}, "reason": "FailedValidation", "message": "neither of sync or async mode is enabled (deleted false)"}
2022-08-06T20:43:55.439+0530    INFO    controllers.VolumeReplicationGroup      Updated VRG Status VolRep pvccount (0), VolSync pvccount(0) DataReady Condition (&Condition{Type:DataReady,Status:False,ObservedGeneration:2,LastTransitionTime:2022-08-06 20:43:55 +0530 IST,Reason:Error,Message:VolumeReplicationGroup mode is invalid,})    {"VolumeReplicationGroup": "default/asdf", "rid": "62208185-4b37-4fb6-b12d-8df78fbf58a8"}
2022-08-06T20:43:55.439+0530    INFO    controllers.VolumeReplicationGroup      VolRep count 0, VolSync count 0 {"VolumeReplicationGroup": "default/asdf", "rid": "62208185-4b37-4fb6-b12d-8df78fbf58a8"}
2022-08-06T20:43:55.439+0530    INFO    controllers.VolumeReplicationGroup      Exiting reconcile loop  {"VolumeReplicationGroup": "default/asdf", "rid": "62208185-4b37-4fb6-b12d-8df78fbf58a8"}
2022-08-06T20:43:55.439+0530    INFO    controllers.VolumeReplicationGroup      Entering reconcile loop {"VolumeReplicationGroup": "default/asdf", "rid": "eb1c4796-27cf-4671-8998-67ff7d838ebe"}
2022-08-06T20:43:55.440+0530    ERROR   controllers.VolumeReplicationGroup      Invalid request detected        {"VolumeReplicationGroup": "default/asdf", "rid": "eb1c4796-27cf-4671-8998-67ff7d838ebe", "error": "neither of sync or async mode is enabled (deleted false)"}

Reply:
Thanks for posting the logs, it is very helpful.

So "rid" is reconsile event id and not reconsiler id - we will get new id for every event.